pkg/collector/corechecks/containers/containerd_events.go

```diff
@@ -0,0 +1,294 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build containerd
+
+package containers
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	ctrUtil "github.com/DataDog/datadog-agent/pkg/util/containerd"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/api/events"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/gogo/protobuf/proto"
+)
+
+//ContainerdEvent contains the timestamp to make sure we flush all events that happened between two checks
+type ContainerdEvent struct {
```

Comment:
Discussion: do you foresee future uses for this type?
In the current logic, we are allocating these to convert them to datadog events just after

Reply:
I wanted to abstract the unmarshaling logic from the subscriber as well as return what matters given the filters. 
It might be premature, but as I initially coded the util with the AD in mind, I figured it'd be easier from another developer's perspective to deal with a well defined struct than with a `*type.Any` that might lack consistency